language/move-vm/runtime/src/data_cache.rs

```diff
@@ -214,14 +187,9 @@ impl<'r, 'l, S: MoveStorage> DataStore for TransactionDataCache<'r, 'l, S> {
                 Ok(None) => GlobalValue::none(),
                 Err(err) => {
                     let msg = format!("Unexpected storage error: {:?}", err);
-                    // REVIEW: better way to get info out of a PartialVMError?
-                    let (_old_status, _old_sub_status, _old_message, indices, offsets) =
-                        err.all_data();
                     return Err(
                         PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
-                            .with_message(msg)
-                            .at_indices(indices)
```

Comment:
We never actually get any location info here. This trait is always implemented by a type outside the VM that doesn't have any access to VM-internal info like bytecode indexes. E.g., here's the DiemDB implementation, which uses a dummy value: https://github.com/diem/diem/blob/main/language/diem-vm/src/data_cache.rs#L148.

Reply:
In the long term, I think it would probably make sense to have the VM take a `MoveResolver<E>` rather than a `MoveResolver<VMError>` to prevent implementations from having to create useless `VMError`'s, but that refactoring is outside the scope of this PR.